packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -973,13 +935,25 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
     private async writeSummaryTree(
         parentHandle: string | undefined,
         tree: api.ISummaryTree,
-        depth: number = 0): Promise<{ result: ISnapshotResponse, blobsShaToPathCacheLatest?: Map<string, string> }> {
-        // Wait for all pending hashes to complete before using them in convertSummaryToSnapshotTree
-        await Promise.all(this.blobsCachePendingHashes.values());
+        depth: number = 0): Promise<{
+            result: ISnapshotResponse,
+            blobsShaToPathCacheLatest: Map<string, string>,
```

Comment:
Given that we operate these three caches together, should we put then into object with defined interface, and just use that type to return / assign (instead of doing it per object)?

Reply:
The code would be more readable IMHO

Reply:
Made an interface for these 3 caches.